src/maggma/stores/azure.py

```diff
@@ -0,0 +1,588 @@
+    ) -> List:
+        """
+        Get all distinct values for a field
+
+        Args:
+            field: the field(s) to get distinct values for
+            criteria: PyMongo filter for documents to search in
+        """
+        # Index is a store so it should have its own distinct function
+        return self.index.distinct(field, criteria=criteria)
+
+    def groupby(
+        self,
+        keys: Union[List[str], str],
+        criteria: Optional[Dict] = None,
+        properties: Union[Dict, List, None] = None,
+        sort: Optional[Dict[str, Union[Sort, int]]] = None,
+        skip: int = 0,
+        limit: int = 0,
+    ) -> Iterator[Tuple[Dict, List[Dict]]]:
+        """
+        Simple grouping function that will group documents
+        by keys.
+
+        Args:
+            keys: fields to group documents
+            criteria: PyMongo filter for documents to search in
+            properties: properties to return in grouped documents
+            sort: Dictionary of sort order for fields. Keys are field names and
+                values are 1 for ascending or -1 for descending.
+            skip: number documents to skip
+            limit: limit on total number of documents returned
+
+        Returns:
+            generator returning tuples of (dict, list of docs)
+        """
+        return self.index.groupby(
+            keys=keys,
+            criteria=criteria,
+            properties=properties,
+            sort=sort,
+            skip=skip,
+            limit=limit,
+        )
+
+    def ensure_index(self, key: str, unique: bool = False) -> bool:
+        """
+        Tries to create an index and return true if it suceeded
+
+        Args:
+            key: single key to index
+            unique: Whether or not this index contains only unique keys
+
+        Returns:
+            bool indicating if the index exists/was created
+        """
+        return self.index.ensure_index(key, unique=unique)
+
+    def update(
+        self,
+        docs: Union[List[Dict], Dict],
+        key: Union[List, str, None] = None,
+        additional_metadata: Union[str, List[str], None] = None,
+    ):
+        """
+        Update documents into the Store
+
+        Args:
+            docs: the document or list of documents to update
+            key: field name(s) to determine uniqueness for a
+                 document, can be a list of multiple fields,
+                 a single field, or None if the Store's key
+                 field is to be used
+            additional_metadata: field(s) to include in the blob store's metadata
+        """
+
+        if self.container is None or self.service is None:
+            raise RuntimeError("The store has not been connected")
+
+        if not isinstance(docs, list):
+            docs = [docs]
+
+        if isinstance(key, str):
+            key = [key]
+        elif not key:
+            key = [self.key]
```

Comment:
It is the same in aws, but if we initialize the store with self.key as a list of multiple fields, this won't work right ?